app/mongo_db_connection.py

```diff
@@ -1170,3 +1170,43 @@ def delete_items_in_folder(item_id, folder_id, item_type):
         }
     )
     return post_to_data_service(payload)
+
+def bulk_query_template_collection_modified(options, status=None):
```

Comment:
This is what we have been avoiding, Stick to using the bulk query template collection. pass status in the options object.